nntrainer/tensor/hgemm/hgemm_kernel/hgemm_kernel_8x16_experimental.cpp

```diff
@@ -0,0 +1,799 @@
+    vst1q_f32(c + 5 * ldc + 8 + 4,                                             \
+              vaddq_f32(vld1q_f32(c + 5 * ldc + 8 + 4),                        \
+                        vcvt_f32_f16(vget_high_f16(v104_111))));               \
+                                                                               \
+    vst1q_f32(c + 6 * ldc, vaddq_f32(vld1q_f32(c + 6 * ldc),                   \
+                                     vcvt_f32_f16(vget_low_f16(v48_55))));     \
+    vst1q_f32(c + 6 * ldc + 4,                                                 \
+              vaddq_f32(vld1q_f32(c + 6 * ldc + 4),                            \
+                        vcvt_f32_f16(vget_high_f16(v48_55))));                 \
+                                                                               \
+    vst1q_f32(c + 6 * ldc + 8,                                                 \
+              vaddq_f32(vld1q_f32(c + 6 * ldc + 8),                            \
+                        vcvt_f32_f16(vget_low_f16(v112_119))));                \
+    vst1q_f32(c + 6 * ldc + 8 + 4,                                             \
+              vaddq_f32(vld1q_f32(c + 6 * ldc + 8 + 4),                        \
+                        vcvt_f32_f16(vget_high_f16(v112_119))));               \
+                                                                               \
+    vst1q_f32(c + 7 * ldc, vaddq_f32(vld1q_f32(c + 7 * ldc),                   \
+                                     vcvt_f32_f16(vget_low_f16(v56_63))));     \
+    vst1q_f32(c + 7 * ldc + 4,                                                 \
+              vaddq_f32(vld1q_f32(c + 7 * ldc + 4),                            \
+                        vcvt_f32_f16(vget_high_f16(v56_63))));                 \
+                                                                               \
+    vst1q_f32(c + 7 * ldc + 8,                                                 \
+              vaddq_f32(vld1q_f32(c + 7 * ldc + 8),                            \
+                        vcvt_f32_f16(vget_low_f16(v120_127))));                \
+    vst1q_f32(c + 7 * ldc + 8 + 4,                                             \
+              vaddq_f32(vld1q_f32(c + 7 * ldc + 8 + 4),                        \
+                        vcvt_f32_f16(vget_high_f16(v120_127))));               \
+  } while (0)
+
+void hgemm_kernel_8x16_experimental(unsigned int M, unsigned int N,
+                                    unsigned int K, __fp16 *sa, __fp16 *sb,
+                                    float *sc, unsigned int ldc) {
+  assert(M > 0 && N > 0 && K > 0);
+  assert(M % 8 == 0 && N % 16 == 0 && K % 4 == 0);
+
+  __fp16 *a = sa, *b = sb;
+  float *c = sc;
+  unsigned int i, j, l;
+  unsigned int K4 = (K >> 2) << 2;
+  unsigned int K8 = (K >> 3) << 3;
+  unsigned int K16 = (K >> 4) << 4;
+  unsigned int K32 = (K >> 5) << 5;
+  unsigned int K64 = (K >> 6) << 6;
```

Comment:
how about using newly added util func `get_prev_mltpl_of_2p_n`?

```suggestion
  unsigned int K4 = get_prev_mltpl_of_2p_n(K, 2);
  unsigned int K8 = get_prev_mltpl_of_2p_n(K, 3);
  unsigned int K16 = get_prev_mltpl_of_2p_n(K, 4);
  unsigned int K32 = get_prev_mltpl_of_2p_n(K, 5);
  unsigned int K64 = get_prev_mltpl_of_2p_n(K, 6);
```